packages/aws-cdk-lib/pipelines/lib/codepipeline/codepipeline-source.ts

```diff
@@ -424,15 +424,32 @@ class CodeStarConnectionSource extends CodePipelineSource {
   constructor(repoString: string, readonly branch: string, readonly props: ConnectionSourceOptions) {
     super(repoString);
 
-    const parts = repoString.split('/');
-    if (Token.isUnresolved(repoString) || parts.length !== 2) {
-      throw new Error(`CodeStar repository name should be a resolved string like '<owner>/<repo>', got '${repoString}'`);
+    if (!this.isValidRepoString(repoString)) {
+      throw new Error(`CodeStar repository name should be a resolved string like '<owner>/<repo>' or '<owner>/<group>/<repo>', got '${repoString}'`);
     }
+
+    const parts = repoString.split('/');
+
     this.owner = parts[0];
-    this.repo = parts[1];
+    this.repo = parts.slice(1).join('/');
     this.configurePrimaryOutput(new FileSet('Source', this));
   }
 
+  private isValidRepoString(repoString: string) {
+    if (Token.isUnresolved(repoString)) {
+      return false;
+    }
+
+    const parts = repoString.split('/');
+
+    if (parts.length < 2) {
```

Comment:
According to the GitLab [docs](https://docs.gitlab.com/ee/user/group/subgroups/), subgroups can be nested up to 20 levels. However it is unclear whether the total number of groups considers parent group or not. At the moment, the validation assumes the total number of sub groups is twenty.
For BitBucket or GitHub, I can not find any features related to sub groups.

Reply:
Ok, thanks for the reference. I'm ok with not adding validation for now then since it might be different for github and bitbucket. Could re-visit in the future